### PR TITLE
Shelter Review Creation

### PR DESCRIPTION
**Pull request by SMJ289**
- Create ShelterReviewsController
- Create ShelterReview Model
- Add new and create methods to ShelterReviewsController
- Add ShelterReviews creation form
- Add ShelterReviews to Shelter show page
- Add happy and sad paths for ShelterReview creation
- [x] All tests passing

### CODE REVIEW

File: app/controllers/shelter_reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class ShelterReviewsController < ApplicationController
+  def new
+    @shelter = Shelter.find(params[:id])
+  end
+
+  def create
+    @shelter = Shelter.find(params[:id])
```
Comment (brian-greeson) [inline app/controllers/shelter_reviews_controller.rb]: We should move this to inside the sad_path line21 so it doesn't make unnecessary db calls.

File: app/controllers/shelter_reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      image: shelter_review_params[:image],
+      content: shelter_review_params[:content],
+      rating: shelter_review_params[:rating],
+      shelter_id: params[:id]
+      })
+
+    if shelter_review.save
+      redirect_to "/shelters/#{params[:id]}"
+    else
+      flash.now[:ding_dong] = "Please include a title and content for your review."
```
Comment (brian-greeson) [inline app/controllers/shelter_reviews_controller.rb]: Do we need to test if the flash.now is clearing?

File: app/models/shelter.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class Shelter < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
   has_many :pets, dependent: :destroy
+  has_many :shelter_reviews, dependent: :destroy
```
Comment (brian-greeson) [inline app/models/shelter.rb]: This great!

File: app/views/shelter_reviews/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<h2>
+  <%= form_tag "/shelters/#{@shelter.id}/reviews" do %>
```
Comment (brian-greeson) [inline app/views/shelter_reviews/new.html.erb]: We could also use a form_with syntax

File: app/controllers/shelter_reviews_controller.rb (hunk 2 above)
Comment (SMJ289) [reply app/controllers/shelter_reviews_controller.rb]: That might be useful but I am not sure how to do that.

File: app/views/shelter_reviews/new.html.erb (hunk 4 above)
Comment (SMJ289) [reply app/views/shelter_reviews/new.html.erb]: That is true but the rest of the forms follow this format so I think it makes the most sense to be consistent. We can decide on our final approach during a refactor.

File: app/controllers/shelter_reviews_controller.rb (hunk 1 above)
Comment (SMJ289) [reply app/controllers/shelter_reviews_controller.rb]: Good call. I went ahead and made that change. The tests are all still passing so the sad path should still be functional in this configuration.

**Pull request merged**